flink-connectors/flink-connector-hbase-2.2/src/test/java/org/apache/flink/connector/hbase2/source/HBaseRowDataAsyncLookupFunctionTest.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.connector.hbase2.source;
+
+import org.apache.flink.connector.hbase.options.HBaseLookupOptions;
+import org.apache.flink.connector.hbase.util.HBaseTableSchema;
+import org.apache.flink.connector.hbase.util.PlannerType;
+import org.apache.flink.connector.hbase2.util.HBaseTestBase;
+import org.apache.flink.table.api.DataTypes;
+import org.apache.flink.table.api.TableSchema;
+import org.apache.flink.table.data.RowData;
+
+import org.apache.flink.shaded.guava18.com.google.common.collect.Lists;
+
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.Collectors;
+
+import static org.junit.Assert.assertEquals;
+
+/** Test suite for {@link HBaseRowDataAsyncLookupFunction}. */
+public class HBaseRowDataAsyncLookupFunctionTest extends HBaseTestBase {
```

Comment:
Could we add an ITCase for `AsyncLookupFunction`  like  `HBaseConnectorITCase.testHBaseLookupTableSource` ?

Reply:
Yes, I will do it.

Reply:
> Could we add an ITCase for `AsyncLookupFunction` like `HBaseConnectorITCase.testHBaseLookupTableSource` ?

@leonardBang  I tried to run this method, but an exception was raised. Is this a bug?
Detailed exception information：
`java.lang.NoClassDefFoundError: com.google.common.base.MoreObjects
	at org.apache.calcite.config.CalciteSystemProperty.loadProperties(CalciteSystemProperty.java:404)
	at org.apache.calcite.config.CalciteSystemProperty.<clinit>(CalciteSystemProperty.java:47)
	at org.apache.calcite.util.Util.<clinit>(Util.java:152)
	at org.apache.calcite.sql.type.SqlTypeName.<clinit>(SqlTypeName.java:142)
	at org.apache.calcite.sql.type.SqlTypeFamily.getTypeNames(SqlTypeFamily.java:163)
	at org.apache.calcite.sql.type.ReturnTypes.<clinit>(ReturnTypes.java:127)
	at org.apache.calcite.sql.SqlSetOperator.<init>(SqlSetOperator.java:45)
	at org.apache.calcite.sql.fun.SqlStdOperatorTable.<clinit>(SqlStdOperatorTable.java:97)
	at org.apache.calcite.sql2rel.StandardConvertletTable.<init>(StandardConvertletTable.java:101)
	at org.apache.calcite.sql2rel.StandardConvertletTable.<clinit>(StandardConvertletTable.java:91)
	at org.apache.calcite.tools.Frameworks$ConfigBuilder.<init>(Frameworks.java:234)
	at org.apache.calcite.tools.Frameworks$ConfigBuilder.<init>(Frameworks.java:215)
	at org.apache.calcite.tools.Frameworks.newConfigBuilder(Frameworks.java:199)
	at org.apache.flink.table.planner.delegation.PlannerContext.createFrameworkConfig(PlannerContext.java:135)
	at org.apache.flink.table.planner.delegation.PlannerContext.<init>(PlannerContext.java:115)
	at org.apache.flink.table.planner.delegation.PlannerBase.<init>(PlannerBase.scala:113)
	at org.apache.flink.table.planner.delegation.StreamPlanner.<init>(StreamPlanner.scala:49)
	at org.apache.flink.table.planner.delegation.BlinkPlannerFactory.create(BlinkPlannerFactory.java:48)
	at org.apache.flink.table.api.bridge.java.internal.StreamTableEnvironmentImpl.create(StreamTableEnvironmentImpl.java:143)
	at org.apache.flink.table.api.bridge.java.StreamTableEnvironment.create(StreamTableEnvironment.java:113)
	at org.apache.flink.table.api.bridge.java.StreamTableEnvironment.create(StreamTableEnvironment.java:85)`

Reply:
HBaseConnectorITCase can not run in IDE directly because of calss conflicts ，we can use `mvn` to test it, try:
```cd flink-connector-hbase-2.2```
```mvn test -Dtest=org.apache.flink.connector.hbase.HBaseConnectorITCase -Dcheckstyle.skip=true```

Reply:
Thank you very much!